src/polycubed/src/cube.cpp

```diff
@@ -37,14 +43,23 @@ Cube::Cube(const std::string &name, const std::string &service_name,
   for (uint16_t i = 0; i < _POLYCUBE_MAX_PORTS; i++)
     free_ports_.insert(i);
 
-  shadow_ = shadow;
-  span_ = span;
+  if (shadow_) {
+    span_ = span;
```

Comment:
What about the following command? `polycubectl router add r0 shadow=false span=true`
There is not warning, error, just nothing, the span is just ignored. This is not ok.
An explicit error should be thrown in that case

Reply:
Done